### PR TITLE
[ML][Pipelines] Copy early available property for pipeline component output

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR targets to support the following scenario: consume component with early available output and return it as pipeline component output. Now the pipeline component output will also hold the early available property.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py [hunk 1]
```diff
@@ -2581,3 +2581,19 @@ def root_pipeline():
         pipeline_job.settings.default_compute = "cpu-cluster"
         validate_result = pipeline_job._validate()
         assert validate_result.error_messages == {}
+
+    def test_dsl_pipeline_component_early_available_property(self):
+        component_func = load_component("./tests/test_configs/components/streaming_component_basic.yml")
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py:2586]: what's the behavior if we are using a registered component in pipeline job yaml

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py [hunk 2]
```diff
@@ -2581,3 +2581,19 @@ def root_pipeline():
...
+
+        @dsl.pipeline
+        def pipeline_component_func():
+            streaming_node = component_func()
+            return streaming_node.outputs
+
+        @dsl.pipeline
+        def root_pipeline_func():
```
Comment (D-W-) [inline sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py:2594]: What's the behavior when the pipeline has 1 layer? e.g. directly submit the pipeline_component_func() object.

File: sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py [hunk 3]
```diff
@@ -296,7 +297,11 @@ def _map_type(_meta):
                 name=key,
                 data=None,
                 meta=Output(
-                    type=_map_type(meta), description=meta.description, mode=meta.mode, is_control=meta.is_control
+                    type=_map_type(meta),
```
Comment (D-W-) [inline sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py:300]: We may have a unified experience for whether promote node level config(mode/is_control/early_available) to pipeline level in SDK side.
Seems currently we will promote is_control and early_available and won't promote mode. 
We can keep current implementation for now, but I think we need a long term task to discuss this.

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py (hunk 1 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py:2586]: it should be same as anonymous, let me verify and add a test on this.

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py (hunk 2 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py:2594]: Just verify for this, it remains `is_control` and `early_available`, same as in pipeline component.

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py (hunk 1 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py:2586]: I have created a component with early available output, and consume it in a pipeline job yaml.
the created pipeline job output has correct early available output as expected, however, as we do not load the component locally in such workflow, SDK will not resolve the arm id and it seems that we cannot check this property as a test.

**Comment by zhengfeiwang** [comment]
Close this PR for not considering pipeline component scenario for streaming for now.

**Pull request closed without merging**